### PR TITLE
0.7.0 RC3 - bug fix pack

### PR DESCRIPTION
**Pull request by Culprit-CZ**
#### 1) Right

Right click on profile tabs -> Display page

> https://github.com/phpBB-Social-Network/phpBB-Social-Network/issues/5
#### 2) Sort order for opened chatboxes

Correct time for chatbox from <code>$user->data['session_start']</code> to <code>$starttime</code>

> http://phpbbsocialnetwork.com/viewtopic.php?f=59&t=4096

### CODE REVIEW

File: root/socialnet/im.php [hunk 1]
```diff
@@ -253,7 +252,7 @@ function load($mode)
 		 */
 		function openChatBox()
 		{
-			global $db, $template, $user, $phpbb_root_path, $phpEx;
+			global $db, $template, $user, $phpbb_root_path, $phpEx, $starttime;
```
Comment (Culprit-CZ) [inline root/socialnet/im.php:255]: ChatBoxes

File: root/socialnet/im.php [hunk 2]
```diff
@@ -268,16 +267,17 @@ function openChatBox()
 			$this->items['onlineUsers'] = $this->p_master->onlineUsers();
 			$usernameToSQL = $db->sql_escape($usernameTo);
 
-			$sql = "INSERT INTO " . SN_IM_CHATBOXES_TABLE . " (uid_from, uid_to, username_to, starttime) VALUES ( {$user->data['user_id']}, {$userTo}, '{$usernameToSQL}', " . $user->data['session_start'] . ")";
+			$sql = "INSERT INTO " . SN_IM_CHATBOXES_TABLE . " (uid_from, uid_to, username_to, starttime) VALUES ( {$user->data['user_id']}, {$userTo}, '{$usernameToSQL}', " . $starttime . ")";
```
Comment (Culprit-CZ) [inline root/socialnet/im.php:270]: ChatBoxes

File: root/socialnet/im.php [hunk 3]
```diff
@@ -268,16 +267,17 @@ function openChatBox()
...
 			$return = $db->sql_query($sql);
+
 			if (!$return)
 			{
 				$sql = "UPDATE " . SN_IM_CHATBOXES_TABLE . "
-									SET starttime = '{$user->data['session_start']}'
-										WHERE uid_from = '{$user->data['user_id']}'
-											AND uid_to = '{$userTo}'
-											AND username_to = '{$usernameToSQL}'";
+			  SET starttime = '{$starttime}'
```
Comment (Culprit-CZ) [inline root/socialnet/im.php]: ChatBoxes

File: root/socialnet/im.php [hunk 4]
```diff
@@ -684,11 +683,10 @@ function onlineList()
 				}
 
 				$in_group = array_diff_key($users, $in_group_added);
-				$array_first = array_shift(array_values($groups));
 				$template->assign_block_vars('sn_im_online_ufg', array(
 					'GID'	 => '0',
 					'NAME'	 => $user->lang['IM_GROUP_UNDECIDED'],
-					'HIDDEN' => $array_first['collapse'],
```
Comment (Culprit-CZ) [inline root/socialnet/im.php:691]: Group Warning problem

**Comment by Kamahl19** [comment]
nehovorili sme o tom ze kazda zmena ma mat vlastny branch a vlastny PR?

File: root/socialnet/im.php [hunk 5]
```diff
@@ -175,70 +174,70 @@ function load($mode)
...
-
-				case 'snImTyping':
-					$this->_snImTyping();
-					return;
-					break;
-
-				case 'snImDisplaySmilies':
-					$this->_displaySmilies();
-					break;
+			case 'startIM':
```
Comment (senky) [inline root/socialnet/im.php]: Why is this moved 1 tab to right?

Also, `break` should be on the same level as `case`:

```
case 'somecase':
   do_stuff()...
break;
```

Whatsmore, you do not use tabs, but spaces. Please, switch it to tabs.

File: root/socialnet/im.php (hunk 3 above)
Comment (senky) [inline root/socialnet/im.php]: As mentioned above, give it more indention and switch to tabs, not spaces.

Also, `'{$starttime}'` is not recommended. Better to use:

```
SET starttime = '" . $starttime . "'
```

or at least:

```
SET starttime = '$starttime'
```

File: root/socialnet/im.php [hunk 6]
```diff
@@ -520,8 +519,8 @@ function _startIM($mode = '')
 
 			// Close chatboxed opened before logging in
 			$sql = "DELETE FROM " . SN_IM_CHATBOXES_TABLE . "
-								WHERE uid_from = '{$user->data['user_id']}'
-									AND starttime < '{$user->data['session_start']}'";
+					WHERE uid_from = '{$user->data['user_id']}'
```
Comment (senky) [inline root/socialnet/im.php:522]: indent + spaces

File: root/socialnet/im.php [hunk 7]
```diff
@@ -529,10 +528,10 @@ function _startIM($mode = '')
 
 			// Load opened chatboxes
 			$sql = "SELECT *
-								FROM " . SN_IM_CHATBOXES_TABLE . "
-									WHERE uid_from = '{$user->data['user_id']}'
-										AND starttime >= '{$user->data['session_start']}'
-								ORDER BY starttime ASC";
+					FROM " . SN_IM_CHATBOXES_TABLE . "
```
Comment (senky) [inline root/socialnet/im.php:531]: indent + spaces

File: root/socialnet/im.php [hunk 8]
```diff
@@ -684,11 +683,10 @@ function onlineList()
 				}
 
 				$in_group = array_diff_key($users, $in_group_added);
-				$array_first = array_shift(array_values($groups));
 				$template->assign_block_vars('sn_im_online_ufg', array(
 					'GID'	 => '0',
 					'NAME'	 => $user->lang['IM_GROUP_UNDECIDED'],
-					'HIDDEN' => $array_first['collapse'],
+					'HIDDEN' => $groups[0]['collapse'],
```
Comment (senky) [inline root/socialnet/im.php:689]: You did not pull-ed from upstream before creation of new branch, right? Please, pull from develop or change this so it will not be part of this PR.

File: root/socialnet/js/m.profile.js [hunk 9]
```diff
@@ -105,6 +111,8 @@
 
 			        }
 			    }).removeAttr('style');
+			    
+			    console.log(par_URL);
```
Comment (senky) [inline root/socialnet/js/m.profile.js:115]: huh2?

File: root/socialnet/js/m.profile.js [hunk 10]
```diff
@@ -57,6 +62,7 @@
 				        if ($.sn.getCookie('sn-up-profileTab') == 0) { return false; }
 			        },
 			        select : function(e, ui) {
+			        	console.log(e);
```
Comment (senky) [inline root/socialnet/js/m.profile.js:65]: huh1?

File: root/socialnet/profile.php [hunk 11]
```diff
@@ -128,9 +130,26 @@ function load($mode, $user_id)
...
-				header('Content-type: text/html; charset=UTF-8');
-				die($content);
+
+				if ( $fullpage == 'false')
+				{
+					$content = $this->p_master->get_page("sn_{$call_mode}");
+					header('Content-type: text/html; charset=UTF-8');
+					die($content);
+				}
+				else
```
Comment (senky) [inline root/socialnet/profile.php:140]: Please, do not just write, that tab should not be opened by right click. Simply load all data needed and display full page, or redirect user to specified page, where wanted tab will be displayed (what is probably not covered in the code right now).

**Comment by senky** [comment]
@Kamahl19: ano, tiez neviem preco to nie je rozdelene na 2 PR, jeden o "right click" a druhy o "sort chatboxes"

File: root/socialnet/im.php (hunk 3 above)
Comment (Culprit-CZ) [inline root/socialnet/im.php]: ok

**Comment by Culprit-CZ** [comment]
Panove, bud to schvalte nebo to zamitnete, me je to sumak, ale delejte. Zde uz neni cas na prodlevy

File: root/socialnet/profile.php (hunk 11 above)
Comment (Culprit-CZ) [inline root/socialnet/profile.php:140]: Neni sance zobrazit prislusny tab. Muselo by se urpravit cookie a nastavit cookie pro zutne TABy Browseru a ruzne TABy profilu.
A na to uz nemam. Takze toto je prvni reseni, kterym se to da vyresit. Zpusob zobrazeni cele stranky nechavame do 0.7.1

File: root/socialnet/js/m.profile.js (hunk 10 above)
Comment (Culprit-CZ) [inline root/socialnet/js/m.profile.js:65]: Tys nikdy nic nezapomel :D

File: root/socialnet/js/m.profile.js (hunk 9 above)
Comment (Culprit-CZ) [inline root/socialnet/js/m.profile.js:115]: Jo to tam nema co delat

File: root/socialnet/im.php (hunk 8 above)
Comment (Culprit-CZ) [inline root/socialnet/im.php:689]: TVL, ja hledam to nastaveni pro NETBEANS a nemuzu jej najit. NEmam cas jeste vsechno kontrolovat.

File: root/socialnet/im.php (hunk 5 above)
Comment (Culprit-CZ) [inline root/socialnet/im.php]: Tak mi priprav nastaveni NetBeans :D

File: root/socialnet/js/m.profile.js (hunk 10 above)
Comment (senky) [inline root/socialnet/js/m.profile.js:65]: jasne ze ano :) prave preto som tak apeloval, aby sme posielali PR a mergovali az ked dalsi dvaja developeri budu suhlasit, aby sa na nic nezabudlo ;)

File: root/socialnet/profile.php (hunk 11 above)
Comment (senky) [inline root/socialnet/profile.php:140]: ok

**Comment by senky** [comment]
Zatvorim to bez mergnutia, ak to nema byt opravene. Ale poslem novy PR, kde to spravim ja, so vsetkym potrebnym (tabs, coding guidelines dodrzane, ...)

**Comment by Culprit-CZ** [comment]
Tak to senky rozdelil do 2 PR. Tak to zaviram, protoze je to zbytecne
